de.dashup.model/src/main/java/de/dashup/model/service/DashupService.java

```diff
@@ -322,31 +284,105 @@ public void updateSection(User user, String sectionName, int sectionId, int pred
             values.put("section_name", sectionName);
         }
         values.put("predecessor_id", predecessor);
-        values.put("successor_id", successor);
 
         if (!values.isEmpty()) {
             this.database.update(Database.Table.USER_SECTIONS, whereParameters, values);
         }
     }
 
-    public void deleteSection(User user, int section_id) throws SQLException {
+    private void deleteSection(User user, int section_id) throws SQLException {
+        deletePanelsOfSection(section_id);
         Map<String, Object> whereParameters = new HashMap<>();
         whereParameters.put("section_id", section_id);
         whereParameters.put("user_id", user.getId());
         database.delete(Database.Table.USER_SECTIONS, whereParameters);
     }
 
-    public void addSection(User user, String sectionName, int predecessor, int successor) throws SQLException {
-        if (sectionName == null) {
-            sectionName = "New Section";
-        }
+    private void deletePanelsOfSection(int section_id) throws SQLException {
+        Map<String, Object> whereParameters = new HashMap<>();
+        whereParameters.put("section_id", section_id);
+        database.delete(Database.Table.SECTIONS_PANELS, whereParameters);
+
+    }
+
+    private int addSection(User user, String sectionName, int order) throws SQLException, NumberFormatException {
 
         Map<String, Object> values = new HashMap<>();
-        values.put("section_name", sectionName);
+
+        values.put("section_name", Objects.requireNonNullElse(sectionName, "New Section"));
         values.put("user_id", user.getId());
-        values.put("predecessor_id", predecessor);
-        values.put("successor_id", successor);
+        values.put("predecessor_id", order);
 
         this.database.insert(Database.Table.USER_SECTIONS, values);
+
+        JSONObject jsonObject = this.database.get(Database.Table.USER_SECTIONS, values).getJSONObject(0);
```

Comment:
There is a method to get the latest id called `Database#getLatestId(Table table)`. You don't need to execute another `Database#get`.